aait/backend/group-2B/Application/Features/Interactions/Handlers/Commands/DeleteCommentInteractionCommandHandler.cs

```diff
@@ -0,0 +1,46 @@
+using AutoMapper;
+using SocialSync.Application.Features.Interactions.Requests.Commands;
+using MediatR;
+using SocialSync.Application.Contracts.Persistence;
+using System.ComponentModel.DataAnnotations;
+using SocialSync.Application.DTOs.InteractionDTOs.CommentDTOs.Validator;
+
+namespace SocialSync.Application.Features.Interactions.Handlers.Commands;
+
+public class DeleteCommentInteractionCommandHandler
+    : IRequestHandler<DeleteCommentInteractionCommand, Unit>
+{
+   
+    private readonly IMapper _mapper;
+    private readonly IUnitOfWork _unitOfWork;
+
+    public DeleteCommentInteractionCommandHandler(
+        IMapper mapper,
+        IUnitOfWork unitOfWork
+    )
+    {
+        _mapper = mapper;
+        _unitOfWork = unitOfWork;
+    }
+
+    public async Task<Unit> Handle(
+        DeleteCommentInteractionCommand command,
+        CancellationToken cancellationToken
+    )
+    {
+        var validator = new DeleteCommentDtoValidator(_unitOfWork);
+        var validationResult = await validator.ValidateAsync(command.DeleteCommentInteractionDto);
+
+        if (!validationResult.IsValid)
+        {
+            string errors = string.Join(", ", validationResult.Errors.Select(e => e.ErrorMessage));
+            throw new ValidationException(errors);
+        }
+        else
+        {
+            var foundComment = await _unitOfWork.InteractionRepository.GetAsync(command.DeleteCommentInteractionDto.Id);
+            await _unitOfWork.InteractionRepository.DeleteAsync(foundComment);
```

Comment:
After deleting the comment, I noticed that the changes are not being saved to the data store. This can lead to data inconsistencies and unexpected behavior.

To resolve this, please make sure to add a call to the SaveChanges() method of the _unitOfWork after deleting the comment. This will ensure that the changes made to the entities are properly saved and persisted.

Possible change would be :
`
 await _unitOfWork.InteractionRepository.DeleteAsync(foundComment);

            if (await _unitOfWork.SaveAsync() == 0)
            {
                throw new Exception ("Error while saving changes");
            }

            return Unit.Value;
`
Note: It will be appropriate to use another custom Exception class for this case.